liquigraph-spring-boot-starter/src/test/java/org/liquigraph/spring/LiquigraphAutoconfigurationTest.java

```diff
@@ -0,0 +1,196 @@
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class LiquigraphAutoconfigurationTest {
+
+    static {
+        SLF4JBridgeHandler.removeHandlersForRootLogger();
+    }
+
+    @ClassRule
+    public static final Neo4jRule neo4j = new Neo4jRule();
+
+    @After
+    public void prepare() {
+        try (Result ignored = graphDb().execute("MATCH (n) DETACH DELETE n")) { }
+    }
+
+    @Test
+    public void runs_migrations_with_bean_configured_by_properties() {
+        try (AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext()) {
+            loadContext(context, "liquigraph.url=jdbc:neo4j:" + neo4j.httpURI().toString());
+
+            assertThat(context.getBeansOfType(SpringLiquigraph.class))
+                .as("SpringLiquigraph is configured through properties and default settings")
+                .isNotEmpty();
+            assertThatMigrationsHaveRun();
+        }
+    }
+
+    @Test
+    public void runs_migrations_with_changelog_property_alias() {
+        try (AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext()) {
+            loadContext(
+                context,
+                "liquigraph.url=jdbc:neo4j:" + neo4j.httpURI().toString(),
+                "liquigraph.changelog=classpath:/db/liquigraph/changelog.xml"
+            );
+
+            assertThat(context.getBeansOfType(SpringLiquigraph.class))
+            .as("SpringLiquigraph is configured through properties and default settings")
+            .isNotEmpty();
+            assertThatMigrationsHaveRun();
+        }
+    }
+
+    @Test
+    public void runs_migrations_with_bean_configured_by_a_single_data_source() {
+        try (AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext()) {
+            loadContext(context, SingleDataSource.class);
+
+            assertThat(context.getBeansOfType(SpringLiquigraph.class))
+                .as("SpringLiquigraph is configured through a provided Datasource bean")
+                .isNotEmpty();
+            assertThatMigrationsHaveRun();
+        }
+    }
+
+    @Test
+    public void runs_migrations_with_bean_configured_by_Liquigraph_data_source() {
+        try (AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext()) {
+            loadContext(context, MultipleDataSources.class);
+
+            assertThat(context.getBeansOfType(SpringLiquigraph.class))
+                .as("SpringLiquigraph is configured through a provided Datasource bean")
+                .isNotEmpty();
+            assertThatMigrationsHaveRun();
+        }
+    }
+
+    @Test
+    public void does_not_inject_bean_when_disabled() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)